### PR TITLE
Attribution Document generator 

### PR DESCRIPTION
**Pull request by bs-ondem**
Introduces new module for the generation of an *Attribution Document*.

New Dependencies: 
- commons-codec:commons-codec:1.12
- org.slf4j:jcl-over-slf4j:1.7.25
- ch.qos.logback:logback-classic:1.2.3
- org.apache.pdfbox:pdfbox:2.0.16
- org.apache.pdfbox:xmpbox:2.0.16
- com.github.ralfstuckert.pdfbox-layout:pdfbox2-layout:1.0.0

### Request Reviewer
@robertklaus-bosch 

*Type of change*:  
new feature + documentation update 

New tests have been added and build is successful. 

### Checklist
Must:
- [x] All related issues are referenced in commit messages

Optional: *(delete if not applicable)*
- [x] I have provided tests for the changes (if there are changes that need additional tests)
- [x] I have updated the documentation accordingly to my changes

### CODE REVIEW

File: antenna-documentation/src/site/markdown/generators/attribution-document-generator-step.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                <entry key="attribution.doc.productVersion" value=""/>
+                <entry key="attribution.doc.copyrightHolder" value=""/>
+            </configuration>
+        </step>
+    </generators>
+</workflow>
+```
+
+#### Explanation of parameters
+* `attribution.doc.templateKey`: Specifies the bundle, which will be used for the generation of the attribution document.
```
Comment (robertklaus-bosch) [inline antenna-documentation/src/site/markdown/generators/attribution-document-generator-step.md]: If I understand this correctly, it still is a must. Why not make it optional, and the default is hardcoded in Antenna?

Comment (bs-ondem) [reply antenna-documentation/src/site/markdown/generators/attribution-document-generator-step.md]: Good point. After adding the basic template, I forgot to change it in the AttributionDocumentGenerator class.

File: modules/attribution-document/attribution-document-basic-bundle/pom.xml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            <groupId>org.eclipse.sw360.antenna</groupId>
+            <artifactId>attribution-document-core</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
```
Comment (robertklaus-bosch) [inline modules/attribution-document/attribution-document-basic-bundle/pom.xml]: Shouldn't this be a configuration of the parent?

File: modules/attribution-document/attribution-document-core/src/main/java/org/eclipse/sw360/antenna/attribution/document/core/AttributionDocumentGeneratorImpl.java [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+    private static final String PARAGRAPH_MARKUP = "Components are identified by " +
+            "{color:#0000EE}{link[https://github.com/package-url/purl-spec]}package URL (purl){link}{color:#000000}.\n\n";
+
+    private final String documentName;
+    private final File workingDir;
+    private final String templateKey;
+    private final DocumentValues values;
+
+    /**
+     * @param workingDir  (non-null) a writable directory to store intermediate and resulting files
```
Comment (robertklaus-bosch) [inline modules/attribution-document/attribution-document-core/src/main/java/org/eclipse/sw360/antenna/attribution/document/core/AttributionDocumentGeneratorImpl.java]: documentName missing

File: pom.xml [hunk 4]
```diff
@@ -188,6 +189,7 @@
                 <artifactId>commons-io</artifactId>
                 <version>2.6</version>
             </dependency>
+            <!-- TODO: Robert manages v3.9 but antenna manages 3.8.1 -->
```
Comment (robertklaus-bosch) [inline pom.xml]: Do you want to keep these todos? I guess it doesn't hurt to use the older version. I'm just a fan of keeping stuff updated ;)

**Pull request merged**